modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-values-factory/src/main/java/com/liferay/dynamic/data/mapping/form/values/factory/internal/DDMFormValuesFactoryImpl.java

```diff
@@ -382,8 +382,9 @@ protected String getDDMFormFieldParameterValue(
 			DDMFormRendererConstants.DDM_FORM_FIELD_LANGUAGE_ID_SEPARATOR);
 		sb.append(LocaleUtil.toLanguageId(locale));
 
-		if (Validator.isNull(
-				ParamUtil.getString(httpServletRequest, sb.toString()))) {
+		if (ArrayUtil.isEmpty(
```

Comment:
@marcelabc, the bug that you've noticed and originated this change happens because parameters are actually arrays inside the request, so, when the parameter array has only one value as it was believed to be expected before, it's fine to use ParamUtil's getString. The problem is that Grid fields now can have multiple values inside that array. That's why you had to change the approach of verifying if the param is null to check if the parameter actually exists in the request.
The only thing I did in comparison to your original changes was to get all parameters values and check if they are an empty array, meaning `length == 0` or is `null`.

Reply:
Thank you so much for the investigation! I understood now 👍